patient_data_source.py

```diff
@@ -0,0 +1,169 @@
+from abc import ABC, abstractmethod
+import names
+import warnings
+from datetime import datetime
+from enum import Enum
+from fhirclient.models.patient import Patient as FHIR_Patient
+from fhirclient.models.observation import Observation as FHIR_Observation
+
+class Patient(ABC):
+
+	@abstractmethod
+	def get_gender(self) -> str:
+		"""Returns the patient's gender."""		
```

Comment:
what is gender expected to be? a string in `['M', 'F']`? 

Then we should say so in the docstring

Actually the `['M', 'F']` was a mimic-specific encoding so we have no reason to stick to that.

one option is to create an enum with possible values being `male | female | other | unknown` as described [here](https://hl7.org/fhir/patient.html).

I would suggest this. then you have to make `generate_name` do something sensible regardless of what the chosen gender is.

Finally, I think it makes sense to allow this method to not be implemented by a derived `Patient`. It's not a required data item to construct a `Patient` resource later.

Reply:
Would it be better to just return `unknown` for the default gender implementation? Simplifies the code by not having conditionally implemented methods.

Reply:
Sure, that makes sense